src/qz/common/CachedObject.java

```diff
@@ -0,0 +1,97 @@
+package qz.common;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+
+/**
+ * A generic class that encapsulates an object for caching. The cached object
+ * will be refreshed automatically when accessed after its lifespan has expired.
+ *
+ * @param <T> The type of object to be cached.
+ */
+public class CachedObject<T> {
+    public static final long DEFAULT_LIFESPAN = 5000;
+    T cachedObject;
```

Comment:
Consider rename `cachedObject` is not of type `CachedObject`, which is confusing.

Reply:
That is a good point. lastObject? storedObject?

Reply:
I'd be fine with either.